include/llbuild/BuildSystem/BuildExecutionQueue.h

```diff
@@ -114,13 +114,16 @@ class BuildExecutionQueue {
   virtual bool
   executeProcess(QueueJobContext* context,
                  ArrayRef<StringRef> commandLine,
-                 ArrayRef<std::pair<StringRef, StringRef>> environment) = 0;
+                 ArrayRef<std::pair<StringRef, StringRef>> environment,
+                 std::function<void(StringRef)> partialOutputCallback = nullptr) = 0;
 
```

Comment:
Can you break this change out into its own patch, with tests? I would probably call this `hasPartialOutput()`, which is a little more consistent with our existing callback naming conventions.

Also, this change should be added to the documentation comments here.